moquette/broker/src/main/java/io/moquette/broker/Session.java

```diff
@@ -133,8 +156,23 @@ void terminateSession() {
             msg.release();
         }
 
-        // TODO: inflight messages also need to be released, but first we need to prevent
-        // multiple threads from being able to queue messages during shutdown
+        // Release in flight messages (QoS 1/2)
+        List<Integer> inflightKeys = new ArrayList<>(inflightWindow.keySet());
+        inflightKeys.forEach((k) -> {
+            SessionRegistry.EnqueuedMessage msg = inflightWindow.get(k);
+            if (msg != null && inflightWindow.remove(k) == msg) {
```

Comment:
if `inflightWindow.remove(k) != msg` what does that mean?

Reply:
Good catch. This can be removed actually. This code was from before we introduced the reference counting so other threads could messing with this map. But that is no longer the case here